File: src/typescript/encryption/encrypt-operations/multiple-start-encrypt.ts
import { createHiveChain } from "@hiveio/wax";

const { wallet, publicKey1, publicKey2 } = globalThis.snippetsBeekeeperData; /* Import preconfigured beekeeper data specific to snippet examples */

const hiveChain = await createHiveChain();

// Create a transaction
const tx = await hiveChain.createTransaction();

// Start the encryption chain with two keys
tx.startEncrypt(publicKey1, publicKey2)
  .pushOperation({ // Add encrypted operations
    transfer_operation: {
      from: "alice",
      to: "bob",
      amount: hiveChain.hiveCoins(5.100), // Send 5.100 HIVE (Note: Coins, not satoshis)
      memo: "This memo will be encrypted with two keys"
    }
  })
  .stopEncrypt() // Stop the current encryption chain
  .startEncrypt(publicKey1) // Start the encryption chain again, but with one key only
  .pushOperation({ // Add other encrypted operations
    transfer_operation: {
      from: "alice",
      to: "bob",
      amount: hiveChain.hiveCoins(10.050), // Send 10.050 HIVE (Note: Coins, not satoshis)
      memo: "This memo will be encrypted with one key only"
    }
  })
  .stopEncrypt(); // Stop the encryption chain again (optionally)

// Sign and build the transaction
tx.sign(wallet, publicKey1);

console.log(tx.transaction);

File: src/rust/encryption/encrypt_operations/multiple_start_encrypt.rs
use runner::{SnippetsBeekeeperData, snippets_beekeeper_data};
use wax::prelude::*;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    /* Import preconfigured beekeeper data specific to snippet examples */
    let SnippetsBeekeeperData { signer1, public_key1, public_key2, .. } = snippets_beekeeper_data();

    let hive_chain = create_hive_chain(None)?;

    // Create a transaction
    let mut tx = hive_chain.create_transaction(None).await?;

    // Start the encryption chain with two keys
    tx.start_encrypt(&public_key1, Some(&public_key2))
        // Add encrypted operations
        .push_operation(hive_chain.create_operation(Value::TransferOperation(
            Transfer {
                from_account: "alice".into(),
                to_account: "bob".into(),
                // Send 5.100 HIVE (Note: Coins, not satoshis)
                amount: hive_chain.hive_coins(5.100)?,
                memo: "This memo will be encrypted with two keys".into(),
            },
        )))
        .stop_encrypt()? // Stop the current encryption chain
        // Start the encryption chain again, but with one key only
        .start_encrypt(&public_key1, None)
        // Add other encrypted operations
        .push_operation(hive_chain.create_operation(Value::TransferOperation(
            Transfer {
                from_account: "alice".into(),
                to_account: "bob".into(),
                // Send 10.050 HIVE (Note: Coins, not satoshis)
                amount: hive_chain.hive_coins(10.050)?,
                memo: "This memo will be encrypted with one key only".into(),
            },
        )))
        .stop_encrypt()?; // Stop the encryption chain again (optionally)

    // Sign and build the transaction
    tx.perform_operation_encryption(&signer1)?;
    tx.sign(&signer1, &public_key1)?;

    println!("{:#?}", tx.transaction());

    Ok(())
}
